tools/tgstation-server/Reset and Recompile.bat

```diff
@@ -0,0 +1,65 @@
+@echo off
+@title Server Updater
+set HOME = %USERPROFILE%
+
+call config.bat
+call bin\getcurdate.bat
+
+echo This will handle resetting the git repo, recompiling the server, and applying the new version.
+echo Ready?
+
+timeout 120
+
+if exist updating.lk (
+	echo ERROR: A current update script has been detected running. if you know this is a mistake:
+	pause
+	echo Please be double sure that an update script is not currently running, if you think one might be, close this window. otherwise:
+	pause
+)
+
+echo lock>updating.lk
+
+rem if the first arg to nudge.py is not a channel, it is treated as the "source"
+if not defined UPDATE_LOG_CHANNEL set UPDATE_LOG_CHANNEL="UPDATER"
+
+call python bot\nudge.py %UPDATE_LOG_CHANNEL% "Reset local changes job started" >nul 2>nul
+
+cd gitrepo
+git reset -hard
```

Comment:
```
This will handle resetting the git repo, recompiling the server, and applying the new version.
Ready?

Waiting for 117 seconds, press a key to continue ...
error: did you mean `--hard` (with two dashes ?)
##################################
##################################
```